depot/containerstore/instance_identity_handler.go

```diff
@@ -97,6 +103,16 @@ func (h *InstanceIdentityHandler) Update(cred Credential, container executor.Con
 		return err
 	}
 
+	err = h.containerWriter.WriteFile(container.Guid, path.Join(h.containerMountPath, "instance.key"), []byte(cred.Key))
```

Comment:
This doesn't make a lot of sense. The reason the file is renamed (few lines above) is to provide as atomic (as we can) file updates. We try not to write to the file directly to avoid partial reads. I understand that there are two files right now that need to be updated, which isn't entirely atomic, but we might in the future combine them into one file.

Reply:
Would you be happier if we implemented the container writer to do something similar to the instance identity handler, i.e. copy certificate/key to temp files in the container and then rename them in a single exec?

To be honest, I do not think that renaming the files without any synchronisation primitives can be considered atomic. Maybe the correct implementation here would be to pause/freeze the container while doing cert stuff, Garden does not support that though.